grants_tagger_light/training/dataloaders/dataloaders.py

```diff
@@ -0,0 +1,183 @@
+import json
+import numpy as np
+from transformers import AutoTokenizer
+from datasets import Dataset
+
+# TODO refactor the two load funcs into a class
+
+
+def _tokenize(batch, tokenizer: AutoTokenizer, x_col: str):
+    return tokenizer(
+        batch[x_col],
+        padding="max_length",
+        truncation=True,
+        max_length=512,
+    )
+
+
+def _label_encode(batch, mesh_terms_column: str, label2id: dict):
+    batch_labels = []
+    for sample_tags in batch[mesh_terms_column]:
+        sample_labels = []
+        for tag in sample_tags:
+            if tag in label2id:
+                sample_labels.append(label2id[tag])
+        batch_labels.append(sample_labels)
+
+    batch["labels"] = batch_labels
+
+    return batch
+
+
+def _one_hot(batch, label2id: dict):
+    batch["labels"] = [
+        [1 if i in labels else 0 for i in range(len(label2id))]
+        for labels in batch["labels"]
+    ]
+    return batch
+
+
+def load_grants_sample(
```

Comment:
whats the point of this? it seems like a duplicate of the other? if not some refactoring is needed

Reply:
This was used to load a grants sample file, which contains much fewer samples.
Probably not needed anymore so I will just delete it.